doc/manual.rst

```diff
@@ -4166,6 +4183,85 @@ For the above `openArray` and `var` rules, whether the expressions refers to
+
+A *signature* is a static interface. It lists the routines and their shape that
+need to be available for a type in order for said type to satisfy the
+interface.
+
+Signatures are types, which are defined as follows:
+
+.. code-block:: nim
+
+  type S {.signature.} = concept T
+    ...
+
+
+The `concept` keyword must be followed by an identifier, which is the name of
+the placeholder used within the body to refer to the type that is described.
+From this point onwards, this identifier is referred to as the `placeholder`.
+
+The body must be a non-empty list of routine *declarations*. Only `func`:idx:,
+`proc`:idx:, `iterator`:idx:, and `method`:idx: are allowed, `template`:idx:
+and `macro`:idx: are not.
+
+For every routine declaration in the body, at least one parameter must be of
+type `placholder`, `var placeholder`, or `sink placeholder`. Declarations must
+not be explicitly or implicitly generic. For pragmas, only calling conventions
+(except `.closure`) and `tags` and `raises` specifications are allowed.
+
+Special names such as the dot and call operators, as well as hook names are
+*not* allowed for the routine declarations. All other operators (e.g., `[]`,
+`field=`, etc.) are allowed.
+
+In addition, no two declarations must have same name *and* parameter lists.
+Same parameter list means the same number of parameters, in the same order
+and with the same types (parameter names are ignored).
+
+A signature type is a *meta type*. An *applied* signature `A` is a signature
+type `S` applied to a concrete type `T`. The relations and properties are as
+follows:
+* `A` is a concrete type and may be used as such
+* `S` is a super-type of `A`
+* `A` is a super-type of `T`, meaning that an instance of `T` may be used where
+  `A` is expected
+* `A` is l-value convertible to `T`
+
+Application of signature type `S` happens when a value of type `T` is
+implicitly or explicitly converted to `S`.
+
+On application of `S` to `T`, for every routine declaration `x` of `S`, a
+routine `y` must be *visible* in the current context such that:
+* `x` and `y` have the same name
+* after all `placholder`s in the parameter list of `x` are replaced with `T`,
+  the lists of `x` and `y` are the same, ignoring parameter names
+* the return type of `x` and `y` are the same (after `placeholder`s in the
+  return type of `x` were replaced with `T`)
+* the routine kinds of `x` and `y` are the same
+* the calling conventions of `x` and `y` are the same
+* (only if `tags` is provided for `x`) the tags of `x` must be a superset of
+  those of `y`
+* (only if `raises` is provided for `x`) the raised exceptions must be a
+  superset of the one of `y`
+
+If `y` is generic, it is instantiated first. If there are either none, or two
+or more candidates that satisfy the requirements listed above, application
+fails. If sucessful, the resulting application is said to *bind* to `y`.
+
+Two applied signatures are equal are equal iff both are applications of the
```

Comment:
`iff` -> `if`

Reply:
The spelling is deliberate. "iff" is a short form of "if and only if" ([source](https://en.wikipedia.org/wiki/If_and_only_if)).

Reply:
Perhaps it's not shorter if you have to provide this message along with all reading material :stuck_out_tongue:

Reply:
Well, `iff` is already used in other places the manual.

Nonetheless, you're right in that the usage does make the text less accessible; I'll expand it.